Hodge-podge of new features

garthwilliamson opened the pull request:
Hi,

I've added a few new features we have found useful:

* Added ability to set whitelisted IP addresses
* Added a pod disruption budget if there are multiple replicas
* Added priority class name

Let me know if you want individual pull requests for any of these or would prefer some changes. Also feel free to cherry pick as you please.

Cheers,
Garth

Commit:
Added ip whitelisting

Commit:
Add pod disruption budget

This will help prevent nginx going down when node groups
are upgraded

Commit:
Add optional priority class

Because the ingress is useful to get to see cluster statistics, it's useful
to have it set to a higher priority than most other pods

dannyrandall commented on nginx.tf:
```diff
@@ -380,3 +377,19 @@ resource "kubernetes_limit_range" "nginx" {
+resource "kubernetes_pod_disruption_budget" "nginx" {
+  count = var.controller_replicas > 1 ? 1 : 0
+  metadata {
+    name      = "nginx"
```
```suggestion
    name      = var.name
```

dannyrandall commented on nginx.tf:
```diff
@@ -380,3 +377,19 @@ resource "kubernetes_limit_range" "nginx" {
+    namespace = kubernetes_namespace.nginx.metadata.0.name
+  }
+  spec {
+    max_unavailable = "50%"
```
```suggestion
    max_unavailable = var.disruption_budget_max_unavailable
```

dannyrandall commented on variables.tf:
```diff
@@ -35,8 +41,20 @@ variable "lb_ports" {
 }
+
+variable "load_balancer_source_ranges" {
+  description = "The ip whitelist that is allowed to access the nlb"
```
```suggestion
  description = "The ip whitelist that is allowed to access the load balancer"
```

dannyrandall reviewed (changes requested):
Thanks for adding this stuff! Here's a few changes that I think we should make before merging.

garthwilliamson commented:
Hi! I'm discovered a few other issues when I tried to use this to have two seperate ingresses rather than just one with a different name. I'll try to make a couple of pull requests here - I'll turn this one into everything _except_ the name change and create the name change as a new request, as I suspect there will be more conversation to have there around sensible defaults.

garthwilliamson commented:
I've changed this pull request to have only the following changes. When applying on top of master this adds a ["0.0.0.0/32"] to the load balancer config whitelisted ip addresses but makes no other changes (with the default configuration). If replicas is set more than 2, it adds the pod disruption budget as well.

* Added ability to set whitelisted IP addresses
* Added a pod disruption budget if there are multiple replicas
* Added priority class name

dannyrandall reviewed (approved):
lgtm. We'll merge this and figure out #7 before bumping the version.
